devices/wholeBodyDynamics/WholeBodyDynamicsDevice.cpp

```diff
@@ -1531,20 +1719,23 @@ void WholeBodyDynamicsDevice::readContactPoints()
     {
         if( contacts_for_given_subModel[subModel] == 0 )
         {
-            bool ok = measuredContactLocations.addNewContactInFrame(estimator.model(),
-                                                                    subModelIndex2DefaultContact[subModel], //frameIndex in iDynTree
-                                                                    iDynTree::UnknownWrenchContact(iDynTree::FULL_WRENCH,iDynTree::Position::Zero()));
-            if( !ok )
+            //iterate for each added contact frame
+            for(size_t ovFrame = 0; ovFrame < contactFramesNames.size(); ovFrame++)
             {
-                yWarning() << "wholeBodyDynamics: Failing in adding default contact for submodel " << subModel;
+                if(contactFramesIdxValidForSubModel[ovFrame] != iDynTree::FRAME_INVALID_INDEX)
+                {
+                    bool ok = measuredContactLocations.addNewContactInFrame(estimator.model(),
+                                                                            contactFramesIdxValidForSubModel[ovFrame], //frameIndex in iDynTree
+                                                                            unknownExtWrench[ovFrame]);
+                    if( !ok )
+                    {
+                        yWarning() << "wholeBodyDynamics: Failing in adding override contact for submodel " << subModel;
+                    }
+                }
             }
         }
-        /*else
-        {
-             yDebug() << "wholeBodyDynamics: number of contacts in submodel "<<subModel<<" = "<<contacts_for_given_subModel[subModel];
-        }*/
     }
-
+    
```

Comment:
```suggestion

```

Reply:
Thanks for pointing this out @traversaro, do you have any further comments?

Reply:
removed in https://github.com/robotology/whole-body-estimators/pull/48/commits/a91ed5422a088256a65c7c27d838184f94cb28ba